packages/fuels-core/src/abi_encoder.rs

```diff
@@ -119,12 +120,11 @@ impl ABIEncoder {
 
         self.encode_discriminant(*discriminant);
 
-        // The sway compiler has an optimization for enums which have only Units
-        // as variants -- such an enum is encoded only by encoding its
-        // discriminant.
+        // The sway compiler has an optimization for enums which have only Units as variants -- such
+        // an enum is encoded only by encoding its discriminant.
         if !variants.only_units_inside() {
             let param_type = Self::type_of_chosen_variant(discriminant, variants)?;
-            self.encode_enum_padding(variants, param_type);
+            self.encode_enum_with_padding(variants, param_type);
```

Comment:
this function only encodes the enum padding -- thus the name `encode_enum_padding`.

`encode_enum_with_padding` implies that the whole enum with the padding was encoded -- that is done by `encode_enum`.